test/MorphoRewardsActions.t.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.23;
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+import "forge-std/StdUtils.sol";
+import "forge-std/StdMath.sol";
+
+import {CodeJar} from "codejar/src/CodeJar.sol";
+import {IERC4626} from "openzeppelin/interfaces/IERC4626.sol";
+import {IERC20} from "openzeppelin/interfaces/IERC20.sol";
+import {IMorpho, MarketParams, Position} from "src/interfaces/IMorpho.sol";
+import {QuarkWallet} from "quark-core/src/QuarkWallet.sol";
+import {QuarkStateManager} from "quark-core/src/QuarkStateManager.sol";
+
+import {QuarkWalletProxyFactory} from "quark-proxy/src/QuarkWalletProxyFactory.sol";
+
+import {YulHelper} from "./lib/YulHelper.sol";
+import {SignatureHelper} from "./lib/SignatureHelper.sol";
+import {QuarkOperationHelper, ScriptType} from "./lib/QuarkOperationHelper.sol";
+
+import {DeFiScriptErrors} from "src/lib/DeFiScriptErrors.sol";
+
+import "src/DeFiScripts.sol";
+import "src/MorphoScripts.sol";
+
+/**
+ * Tests for Morpho Rewards Claim
+ */
+contract MorphoRewardsActionsTest is Test {
```

Comment:
Nb., but Morpho rewards is so confusing. This test is confusing! The actual work happens off-chain and the claims process is an awkward shuffling dance coordinating so many contracts and addresses. Blah.

In effect this test is asserting that "if Morpho's rewards API gives us these magic numbers, we get out some other magic numbers." Yet I don't see how we can do better! :cry:

Reply:
Yes and people can claim for other wallets too. 
This opens up as we might can provide a service to batch claim for users.